duktape/src/main/java/com/squareup/duktape/Duktape.java

```diff
@@ -125,8 +125,11 @@ private static void checkSignatureSupported(Method method) {
 
   /** Returns true if we support {@code: type} in calls from JavaScript. */
   private static boolean isSupportedType(Class<?> type) {
-    // TODO: support some more types.
-    return void.class.equals(type) || String.class.equals(type);
+    return void.class.equals(type)
+        || boolean.class.equals(type)
+        || int.class.equals(type)
+        || double.class.equals(type)
```

Comment:
Yeah, we could support them all.  I'm not a JS expert, but we'd probably have to do our own limit checking.

e.g. IIRC JavaScript tops out at 53 bit integers or something, so a Java long might be truncated.